### PR TITLE
COMMERCE-6731 - Make Commerce CheckoutStepHelper class as a osgi serv…

### PR DESCRIPTION
**Pull request by janbrychtadotcom**
…ice.

### CODE REVIEW

**Comment by marco-leo** [comment]
@janbrychtadotcom please rebase to fix the conflict

**Review by riccardo-alberti — CHANGES_REQUESTED** [review]
Can we also add integration tests? thnx

File: modules/apps/commerce/commerce-checkout-api/src/main/java/com/liferay/commerce/checkout/helper/CommerceCheckoutStepHelper.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+@ProviderType
+public interface CommerceCheckoutStepHelper {
+
+    String getOrderDetailURL(HttpServletRequest httpServletRequest, CommerceOrder commerceOrder) throws PortalException;
+
+    boolean isActiveBillingAddressCommerceCheckoutStep(HttpServletRequest httpServletRequest, CommerceOrder commerceOrder) throws PortalException;
+
+    boolean isActivePaymentMethodCommerceCheckoutStep(HttpServletRequest httpServletRequest, CommerceOrder commerceOrder) throws PortalException;
+
```
[the remark this answers is not in the document]
Comment (janbrychtadotcom) [reply modules/apps/commerce/commerce-checkout-api/src/main/java/com/liferay/commerce/checkout/helper/CommerceCheckoutStepHelper.java:16]: @riccardo-alberti, here are the reasons for having two arguments for this method ( specified exact commerceOrder) ActiveBillingAddressCommerceCheckoutStep

- getting CommerceOrder from HttpServletRequest differs from the place where you try to get it:
     -  case of Order steps: CommerceOrder commerceOrder =
				(CommerceOrder)httpServletRequest.getAttribute(
						CommerceCheckoutWebKeys.COMMERCE_ORDER);
     - case of Payment Servlet:
             CommerceOrder commerceOrder =
				_commercePaymentHttpHelper.getCommerceOrder(httpServletRequest);
           inside method:
                     long groupId = ParamUtil.getLong(httpServletRequest, "groupId");
		     String uuid = ParamUtil.getString(httpServletRequest, "uuid");
                 
- For the future changes ( in my case of Sintra where I need to change the implementation of this interface) it is absolutely clear what commerceOrder I can work with. I do not need to guess from what httpRequest params/attributes I can get commerceOrder

BTW. I also added commerceOrder as an argument into isActiveBillingAddressCommerceCheckoutStep to have these two methods consistent and again to be clear about the commerceOrder.

File: modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/util/DefaultCommerceCheckoutStepHelper.java [hunk 2]
```diff
@@ -73,57 +75,46 @@ public String getOrderDetailURL(
...
 
-	public boolean isActivePaymentMethodCommerceCheckoutStep(
-			HttpServletRequest httpServletRequest)
-		throws PortalException {
-
-		CommerceOrder commerceOrder =
-			(CommerceOrder)httpServletRequest.getAttribute(
-				CommerceCheckoutWebKeys.COMMERCE_ORDER);
+	@Override
+	public boolean isActivePaymentMethodCommerceCheckoutStep(HttpServletRequest httpServletRequest, CommerceOrder commerceOrder) throws PortalException{
```
[the remark this answers is not in the document]
Comment (janbrychtadotcom) [reply modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/util/DefaultCommerceCheckoutStepHelper.java:104]: done. I pushed new commit - checkout-web: source formatting

[the remark this answers is not in the document]
Comment (janbrychtadotcom) [reply modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/util/DefaultCommerceCheckoutStepHelper.java:104]: I pushed new commit with integration tests

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
@dsitu the SF is complaining about having HttpServletRequest in a api module but there are other examples where this is used. What should we do in this case? thnx

**Comment by dsitu** [comment]
Hey @riccardo-alberti @janbrychtadotcom looking at this [documentation](https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/documentation/api_method_signatures.markdown), there are some exceptions. It might make sense to call `CommerceCheckoutStepHelper.java -> CommerceCheckoutStepHttpHelper.java`, similar to [this](https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/commerce-wish-list-api/src/main/java/com/liferay/commerce/wish/list/util/CommerceWishListHttpHelper.java), otherwise as a last resort we would add a sf [suppression ](https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/source-formatter-suppressions.xml).

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java [hunk 3]
```diff
@@ -340,6 +358,309 @@ public void testGuestUserCheckoutWithGuestCheckoutDisabled()
...
+	@Test
+	public void testIsActiveBillingAddressCommerceCheckoutStepNoAddress()
+		throws Exception {
+
+		frutillaRule.scenario(
+			"Check if BillingAddressCommerceCheckoutStep is active "
+		).given(
+			"An Order"
+		).when(
+			"Order does not have any addresses ( no shipping address, no " +
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java:370]: extra space
```suggestion
			"Order does not have any addresses (no shipping address, no " +
```

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java [hunk 4]
```diff
@@ -340,6 +358,309 @@ public void testGuestUserCheckoutWithGuestCheckoutDisabled()
...
+		);
+
+		CommerceContext commerceContext = new TestCommerceContext(
+			_commerceCurrency, _commerceChannel, _user, _group,
+			_commerceAccount, null);
+
+		HttpServletRequest httpServletRequest = new MockHttpServletRequest();
+
+		httpServletRequest.setAttribute(
+			CommerceWebKeys.COMMERCE_CONTEXT, commerceContext);
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java:383]: Sort
```suggestion
		HttpServletRequest httpServletRequest = new MockHttpServletRequest();

		CommerceContext commerceContext = new TestCommerceContext(
			_commerceCurrency, _commerceChannel, _user, _group,
			_commerceAccount, null);

		httpServletRequest.setAttribute(
			CommerceWebKeys.COMMERCE_CONTEXT, commerceContext);
```

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java [hunk 5]
```diff
@@ -340,6 +358,309 @@ public void testGuestUserCheckoutWithGuestCheckoutDisabled()
...
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java:427]: sort
```suggestion
		HttpServletRequest httpServletRequest = new MockHttpServletRequest();

		CommerceContext commerceContext = new TestCommerceContext(
			_commerceCurrency, _commerceChannel, _user, _group,
			_commerceAccount, null);

		httpServletRequest.setAttribute(
			CommerceWebKeys.COMMERCE_CONTEXT, commerceContext);
```

**Comment by riccardo-alberti** [comment]
@dsitu thanks! @janbrychtadotcom can you please make the suggested change? thx

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java [hunk 6]
```diff
@@ -340,6 +358,309 @@ public void testGuestUserCheckoutWithGuestCheckoutDisabled()
...
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java:483]: sort
```suggestion
		HttpServletRequest httpServletRequest = new MockHttpServletRequest();

		CommerceContext commerceContext = new TestCommerceContext(
			_commerceCurrency, _commerceChannel, _user, _group,
			_commerceAccount, null);

		httpServletRequest.setAttribute(
			CommerceWebKeys.COMMERCE_CONTEXT, commerceContext);
```

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java [hunk 7]
```diff
@@ -340,6 +358,309 @@ public void testGuestUserCheckoutWithGuestCheckoutDisabled()
...
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java:535]: sort
```suggestion
		HttpServletRequest httpServletRequest = new MockHttpServletRequest();

		CommerceContext commerceContext = new TestCommerceContext(
			_commerceCurrency, _commerceChannel, _user, _group,
			_commerceAccount, null);

		httpServletRequest.setAttribute(
			CommerceWebKeys.COMMERCE_CONTEXT, commerceContext);
```

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java [hunk 8]
```diff
@@ -340,6 +358,309 @@ public void testGuestUserCheckoutWithGuestCheckoutDisabled()
...
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java:578]: sort
```suggestion
		HttpServletRequest httpServletRequest = new MockHttpServletRequest();

		CommerceContext commerceContext = new TestCommerceContext(
			_commerceCurrency, _commerceChannel, _user, _group,
			_commerceAccount, null);

		httpServletRequest.setAttribute(
			CommerceWebKeys.COMMERCE_CONTEXT, commerceContext);
```

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java [hunk 9]
```diff
@@ -340,6 +358,309 @@ public void testGuestUserCheckoutWithGuestCheckoutDisabled()
...
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java:639]: sort
```suggestion
		HttpServletRequest httpServletRequest = new MockHttpServletRequest();

		CommerceContext commerceContext = new TestCommerceContext(
			_commerceCurrency, _commerceChannel, _user, _group,
			_commerceAccount, null);

		httpServletRequest.setAttribute(
			CommerceWebKeys.COMMERCE_CONTEXT, commerceContext);
```

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java [hunk 10]
```diff
@@ -486,6 +807,45 @@ public void testUserCheckout() throws Exception {
...
+			RandomTestUtil.randomString(), //name
+			RandomTestUtil.randomString(), //description
+			RandomTestUtil.randomString(), //stree1
+			RandomTestUtil.randomString(), //street2
+			RandomTestUtil.randomString(), //street3
+			RandomTestUtil.randomString(), //city
+			RandomTestUtil.randomString(), //zip
+			0, //regionId
+			country.getCountryId(), //countryId
+			RandomTestUtil.randomString(), //phoneNumber
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java:844]: remove comments when not necessary
```suggestion
			RandomTestUtil.randomString(),
			RandomTestUtil.randomString(),
			RandomTestUtil.randomString(),
			RandomTestUtil.randomString(),
			RandomTestUtil.randomString(),
			RandomTestUtil.randomString(),
			RandomTestUtil.randomString(),
			0,
			country.getCountryId(),
			RandomTestUtil.randomString(),
```

**Review by dsitu — CHANGES_REQUESTED** [review]

**Pull request closed without merging**